src/accessibility/gridOutput.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * @module Environment
+ * @submodule Environment
+ * @for p5
+ * @requires core
+ */
+import p5 from '../core/main';
+
+let ingredients = {};
+let background;
+
+//creates html structure
+p5.prototype._createGridOutput = function(cIdT) {
+  let inner =
+    'Grid Output<p id="' +
+    cIdT +
+    'Summary" aria-label="grid output summary"><table id="' +
+    cIdT +
+    'OD" summary="grid output content"></table><ul id="' +
+    cIdT +
+    'SD" aria-label="grid output shape details"></ul>';
+  return inner;
+};
+
+//creates output content
+p5.prototype._updateGridOutput = function(cnvId, ing, bkgr) {
+  ingredients = ing;
+  background = bkgr;
+  let cIdT = cnvId + 'grOut';
+  let innerSDs = this._buildGridSDs(cIdT);
+  let innerSummary = this._buildGrSummary(innerSDs.numShapes);
+  let innerMap = this._buildGrMap(cIdT);
+  if (innerSummary !== document.getElementById(cIdT + 'Summary').innerHTML) {
+    document.getElementById(cIdT + 'Summary').innerHTML = innerSummary;
+  }
+  if (innerMap !== document.getElementById(cIdT + 'OD').innerHTML) {
+    document.getElementById(cIdT + 'OD').innerHTML = innerMap;
+  }
+  if (innerSDs !== document.getElementById(cIdT + 'SD').innerHTML) {
+    document.getElementById(cIdT + 'SD').innerHTML = innerSDs.details;
+  }
+};
+
+//creates spatial grid
+p5.prototype._buildGrMap = function(cId) {
+  let el = 0;
+  let table = '';
+  let cells = Array.apply(null, Array(10)).map(function() {});
+  for (let r in cells) {
+    cells[r] = Array.apply(null, Array(10)).map(function() {});
+  }
+  for (let x in ingredients) {
+    for (let y in ingredients[x]) {
+      el++;
+      if (
+        cells[ingredients[x][y].loc.locY][ingredients[x][y].loc.locX] ===
+        undefined
+      ) {
+        cells[ingredients[x][y].loc.locY][ingredients[x][y].loc.locX] =
+          '<a href="#' +
```

Comment:
It looks like ES6 has some support for [template strings](https://developers.google.com/web/updates/2015/01/ES6-Template-Strings), which would probably make this kind of block easier to read. I can't remember if we're only supporting ES6 compatible browsers at this time or if we're trying to preserve backwards compatibility. @lmccart is it ok to use ES6 features like this?

Reply:
yes we can use ES6! right now it compiles down to ES5 to support all modern browsers, but it's good to write in ES6 so we are future ready

Reply:
rad! thank you!